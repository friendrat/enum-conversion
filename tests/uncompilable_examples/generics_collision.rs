/// In this test, no foreign traits are derived and
/// no foreign types are in the decorated enum.
///
/// However, if the two generic parameters are set
/// to be equal, the enum will contain variants with
/// identical types. It will the be impossible to implement
/// the `TryTo` trait and the rust compiler will complain
/// accordingly about conflicting implementations.
use enum_conversion::prelude::*;

struct Local<T>(T);

/// Note that
/// ```
/// enum Enum<U> {
///     F1(Local<U>),
///     F2(Local<bool>),
/// }
/// ```
/// would suffer the same problems as `U` could
/// be given the type `bool`.
#[EnumConversions]
enum Enum<U, T> {
    F1(Local<U>),
    F2(Local<T>)
}



fn main() {

}