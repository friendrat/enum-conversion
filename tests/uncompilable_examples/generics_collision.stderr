error[E0119]: conflicting implementations of trait `enum_conversion_traits::TryTo<Local<_>>` for type `Enum<_, _>`
  --> tests/uncompilable_examples/generics_collision.rs:22:1
   |
22 | #[EnumConversions]
   | ^^^^^^^^^^^^^^^^^^
   | |
   | first implementation here
   | conflicting implementation for `Enum<_, _>`
   |
   = note: this error originates in the attribute macro `EnumConversions` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `enum_conversion_traits::TryTo<&Local<_>>` for type `&Enum<_, _>`
  --> tests/uncompilable_examples/generics_collision.rs:22:1
   |
22 | #[EnumConversions]
   | ^^^^^^^^^^^^^^^^^^
   | |
   | first implementation here
   | conflicting implementation for `&Enum<_, _>`
   |
   = note: this error originates in the attribute macro `EnumConversions` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `enum_conversion_traits::TryTo<&mut Local<_>>` for type `&mut Enum<_, _>`
  --> tests/uncompilable_examples/generics_collision.rs:22:1
   |
22 | #[EnumConversions]
   | ^^^^^^^^^^^^^^^^^^
   | |
   | first implementation here
   | conflicting implementation for `&mut Enum<_, _>`
   |
   = note: this error originates in the attribute macro `EnumConversions` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `std::convert::From<Local<_>>` for type `Enum<_, _>`
  --> tests/uncompilable_examples/generics_collision.rs:22:1
   |
22 | #[EnumConversions]
   | ^^^^^^^^^^^^^^^^^^
   | |
   | first implementation here
   | conflicting implementation for `Enum<_, _>`
   |
   = note: this error originates in the attribute macro `EnumConversions` (in Nightly builds, run with -Z macro-backtrace for more info)
